misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -139,9 +149,81 @@ public void generateReport(Map<String, ModuleCoverage> moduleCoverageMap,
         }
     }
 
+    /**
+     * Add/Update package level coverage information into the relevant package level lists.
+     *
+     * @param packagePrefix "orgName"/"pakageName" as String
+     * @param packageExecData ExecutionData list for package
+     * @param sessionInfoList SessionInfo list for package
+     * @param coverageBuilder CoverageBuilder after processing this module
+     * @param packageNativeClassCoverageList List of package native IClassCoverage
+     * @param packageBalClassCoverageList List of bal IClassCoverage for package
+     * @param packageSourceCoverageList List of ISourceFileCoverage for package
+     */
+    private void updatePackageLevelCoverage(String packagePrefix, List<ExecutionData> packageExecData,
+                                            List<SessionInfo> sessionInfoList, CoverageBuilder coverageBuilder,
+                                            List<IClassCoverage> packageNativeClassCoverageList,
+                                            List<IClassCoverage> packageBalClassCoverageList,
+                                            List<ISourceFileCoverage> packageSourceCoverageList) {
+        for (ISourceFileCoverage sourceFileCoverage : coverageBuilder.getSourceFiles()) {
+            if (sourceFileCoverage.getPackageName().contains(packagePrefix)) {
+                packageSourceCoverageList.add(sourceFileCoverage);
+            }
+        }
+        for (IClassCoverage classCov : coverageBuilder.getClasses()) {
+            // Store only the package specific bal classes in list
```

Comment:
Shall we explain what is done here?..As to why need to have a condition and why the existing entry is removed in one case.